tests/e2e_test.go

```diff
@@ -0,0 +1,419 @@
+		fmt.Println(string(out))
+		gomega.Expect(err).Should(gomega.BeNil())
+	})
+
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	ginkgo.It("Send Message from A to B", ginkgo.Label("Warp", "SendWarp"), func() {
+		ctx := context.Background()
+
+		// Create a channel to communicate with the goroutine
+		cmdOutput := make(chan string)
+
+		// Run awm relayer binary with config path
+		relayerContext, relayerCancel := context.WithCancel(ctx)
+		cmd := exec.CommandContext(relayerContext, "./build/awm-relayer", "--config-file", relayerConfigPath)
+
+		// Set up a pipe to capture the command's output
+		cmdReader, _ := cmd.StdoutPipe()
+
+		// Start the command
+		err := cmd.Start()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Start a goroutine to read and output the command's stdout
+		go func() {
+			scanner := bufio.NewScanner(cmdReader)
+			for scanner.Scan() {
+				log.Info(scanner.Text())
+			}
+			cmdOutput <- "Command execution finished"
+		}()
+
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		nonceA, err := chainAWSClient.NonceAt(ctx, fundedAddress, nil)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		nonceB, err := chainBWSClient.NonceAt(ctx, fundedAddress, nil)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		log.Info("Packing teleporter message", "nonceA", nonceA, "nonceB", nonceB)
+
+		payload, err = teleporter.PackTeleporterMessage(common.Hash(blockchainIDB), teleporterMessage)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		packedInput, err := warp.PackSendWarpMessage(warp.SendWarpMessageInput{
+			DestinationChainID: common.Hash(blockchainIDB),
+			DestinationAddress: teleporterContractAddress,
+			Payload:            payload,
+		})
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Send a transaction that simulates the Teleporter contract calling sendWarpMessage with a Teleporter message payload
+		tx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainAIDInt,
+			Nonce:     nonceA,
+			To:        &warp.Module.Address,
+			Gas:       200_000,
+			GasFeeCap: big.NewInt(225 * params.GWei),
+			GasTipCap: big.NewInt(params.GWei),
+			Value:     common.Big0,
+			Data:      packedInput,
+		})
+		txSigner := types.LatestSignerForChainID(chainAIDInt)
+		signedTx, err := types.SignTx(tx, txSigner, teleporterKey)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// Sleep for some time to make sure relayer has started up and subscribed.
+		time.Sleep(15 * time.Second)
+		log.Info("Subscribing to new heads on destination chain")
+
+		newHeadsB := make(chan *types.Header, 10)
+		sub, err := chainBWSClient.SubscribeNewHead(ctx, newHeadsB)
+		gomega.Expect(err).Should(gomega.BeNil())
+		defer sub.Unsubscribe()
+
+		log.Info("Sending sendWarpMessage transaction", "destinationChainID", blockchainIDB, "txHash", signedTx.Hash())
+		err = chainAWSClient.SendTransaction(ctx, signedTx)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		log.Info("Waiting for new block confirmation")
```

Comment:
Can you add a comment above this block? Something like "Get the latest block from subnet B"